File: BE/src/main/java/com/example/starbucks/domain/Category.java
package com.example.starbucks.domain;

import java.util.List;
import lombok.Getter;

import javax.persistence.*;

@Entity
@Getter
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(mappedBy = "category")
    private List<CategoryProduct> categoryProducts;

    @Enumerated(EnumType.STRING)
    private CategoryType type;

    private String koreanName;
    private String englishName;
    private String imageUrl;

}

File: Android/app/src/main/java/com/example/starbucks/dto/Category.kt
package com.example.starbucks.dto


import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Category(
    @SerialName("englishName")
    val englishName: String?,
    @SerialName("id")
    val id: Int?,
    @SerialName("imageUrl")
    val imageUrl: String?,
    @SerialName("koreanName")
    val koreanName: String?
)